src/vm/database/marf.rs

```diff
@@ -14,56 +18,67 @@ use chainstate::burn::BlockHeaderHash;
 ///   loop will need to invoke these two methods (begin + commit) outside of the context of the VM.
 ///   NOTE: Clarity will panic if you try to execute it from a non-initialized MarfedKV context.
 ///   (See: vm::tests::with_marfed_environment()) 
-pub struct MarfedKV <S: KeyValueStorage> {
+pub struct MarfedKV {
     chain_tip: BlockHeaderHash,
-    marf: MARF,
+    // the MARF is option'ed, if None, then this KV is "marfless".
+    //    this is used for raw evals, testing --- use cases formerly that used a generic datastore.
+    // functions which _assume_ a MARF present will panic if the MARF is None.
+    marf: Option<MARF>,
     // Since the MARF only stores 32 bytes of value,
     //   we need another storage
-    side_store: S
+    side_store: SqliteConnection
 }
 
-#[cfg(test)]
-pub fn temporary_marf() -> MarfedKV<std::collections::HashMap<String, String>> {
+struct ContractCommitment {
+    pub hash: [u8; 32],
```

Comment:
Should this struct use a specific hash type, like `Sha512Trunc256`?  i.e. Isn't the choice of hash function going to be consensus-critical here, if this struct (as I understand it right now) is what binds the contract code to the chain state?

Reply:
Addressed in 182cdb3